### PR TITLE
doca-dpdk: update the pipeline entry remove handle

### PR DESCRIPTION
**Pull request by jiaweiwsz**
Update the pipeline entry remove function, firstly find
the matched entry based on the pipe_queue, if found, then
free entry and update the numbers of total pipe entries.

Adds the input argument checking and return if NULL
pointer;

Signed-off-by: Jiawei Wang <jiaweiw@nvidia.com>

### CODE REVIEW

**Comment by jiaweiwsz** [comment]
@rbaryanai , Please help to review it code change for doca flow API error handle and update the doca_flow_pipe_rm_entry function implementation.

Thanks.

File: examples/vnet/doca_dpdk.c [hunk 1]
```diff
@@ -1496,6 +1509,22 @@ doca_dpdk_add_pipe_entry(struct doca_flow_pipe *pipe,
...
 }
 
+int doca_dpdk_find_pipe_entry(uint16_t pipe_queue,
+			      struct doca_flow_pipe *pipe,
+			      struct doca_flow_pipe_entry *pipe_entry)
+{
+	struct doca_flow_pipe_entry *entry;
+	int found = 0;
+
+	LIST_FOREACH(entry, &pipe->entry_list[pipe_queue], next) {
```
Comment (rbaryanai) [inline examples/vnet/doca_dpdk.c]: we should never use linked-list for more than few objects.
see my comment below, you don't need to find it.

File: examples/vnet/doca_flow.c [hunk 2]
```diff
@@ -64,10 +63,26 @@ int
...
-	/* doca_dpdk_free_flow(0, entry->pipe_entry); */
-	free(entry);
-	return 0;
+	struct doca_flow_pipe *pipe;
+	int ret = 0;
+
+	if (entry == NULL || entry->flow_pipe == NULL)
+		return -EINVAL;
+	pipe = (struct doca_flow_pipe *)entry->flow_pipe;
+	if (doca_dpdk_find_pipe_entry(pipe_queue, pipe, entry)) {
```
Comment (rbaryanai) [inline examples/vnet/doca_flow.c]: if entry list pointers are not NULL or trahsed.

**Review by rbaryanai — CHANGES_REQUESTED** [review]

File: examples/vnet/doca_dpdk.c (hunk 1 above)
Comment (jiaweiwsz) [reply examples/vnet/doca_dpdk.c]: OK, 
I have one question how to use 'pipe_queue' in doca_flow_pipe_rm_entry() if not need search?
Only need the entry is enough to remove the resource.

File: examples/vnet/doca_flow.c (hunk 2 above)
Comment (jiaweiwsz) [reply examples/vnet/doca_flow.c]: Done.

**Review by rbaryanai — APPROVED** [review]

**Pull request merged**